connectors/sources/graphql.py

```diff
@@ -0,0 +1,333 @@
+
+    def __init__(self, configuration):
+        """Setup the connection to the GraphQL instance.
+
+        Args:
+            configuration (DataSourceConfiguration): Instance of DataSourceConfiguration class.
+        """
+        super().__init__(configuration=configuration)
+        self.graphql_client = GraphQLClient(configuration=configuration)
+
+    def _set_internal_logger(self):
+        self.graphql_client.set_logger(self._logger)
+
+    @classmethod
+    def get_default_configuration(cls):
+        """Get the default configuration for GraphQL.
+
+        Returns:
+            dict: Default configuration.
+        """
+        return {
+            "http_endpoint": {
+                "label": "HTTP URL & Endpoint",
+                "order": 1,
+                "type": "str",
+            },
+            "http_method": {
+                "display": "dropdown",
+                "label": "GET/POST",
+                "options": [
+                    {"label": "GET", "value": "get"},
+                    {"label": "POST", "value": "post"},
+                ],
+                "order": 2,
+                "type": "str",
+                "value": "POST",
+            },
+            "authentication_method": {
+                "display": "dropdown",
+                "label": "Authentication Method",
+                "options": [
+                    {"label": "No Auth", "value": "none"},
+                    {"label": "Basic Auth", "value": "basic"},
+                    {"label": "Bearer Token", "value": "bearer"},
+                ],
+                "order": 3,
+                "type": "str",
+                "value": "none",
+            },
+            "username": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Username",
+                "order": 4,
+                "type": "str",
+            },
+            "password": {
+                "depends_on": [{"field": "authentication_method", "value": "basic"}],
+                "label": "Password",
+                "order": 5,
+                "sensitive": True,
+                "type": "str",
+            },
+            "token": {
+                "depends_on": [{"field": "authentication_method", "value": "bearer"}],
+                "label": "Bearer Token",
+                "order": 6,
+                "sensitive": True,
+                "type": "str",
+            },
+            "graphql_query": {
+                "display": "textarea",
+                "label": "GraphQL Body",
+                "order": 7,
+                "type": "str",
+            },
+            "graphql_variables": {
+                "depends_on": [{"field": "http_method", "value": "post"}],
+                "display": "textarea",
+                "label": "Graphql Variables",
+                "order": 8,
+                "type": "str",
+                "required": False,
+            },
+            "object_list": {
```

Comment:
consider adding tooltips with explanations for configuration fields, I'm not sure what `object_list` mean in this context (tooltips are useful in our UI and also when inspecting the config in the codebase)